### DISCUSSION TITLE
What the Sulakore SDK is used for?

### DISCUSSION THREAD

**Discussion by VitorGPereira97** [ArachisH/Sulakore#33] [Q&A] [answered]
Because the Tanji Software has a Habbo SDK? I'm studing the Tanji software to learn code better.

**Comment by ArachisH** [comment, marked as answer]
The intention of Sulakore is to provide Connection/Crypto/Parsing specific APIs for anything Habbo related as a library(eventually as nuget packages), without any of the interception/MITM stuff in Tanji.

I'm currently moving most of Tanji's logic into another library here https://github.com/ArachisH/Tangine
This library also uses Sulakore, but it provides the Flash/WebAssembly modifcation logic, which I'm planning on using to update HabKit(command line tool).

**Reply by xrevoltsx** [reply]
sulakore doesn't support socks5 anymore? I'm kind of new in this area I tried to find the SOCKS5EndPoint to update an old application and I didn't find it in this new sulakore

**Reply by PaulusParssinen** [reply]
The latest version of the HNode does not offer any proxy support out of the box. Right now that functionality would seem pretty out-of-scope for the library but we may add it back later. I'm planning some major rewriting of the networking stack which in future could make socks4/5 proxies easy to implement. 

Socks5 is not hard to support but it would just bring more complexity to the current HNode implementation. The handshake is not difficult and basically means implementing following branch just for a normal `Socket`:
https://github.com/ArachisH/Tanji/blob/2d2a0e2a0002fd25890cdadbc098ba47ed6508fc/Sulakore/Communication/HNode.cs#L70-L133

Basically establish a normal Socket connection to the proxy, do the Socks5 handshake (you can follow the code above but it can be simplified a lot, it's old code) with the actual endpoint you want the proxy to connect to, and then pass the new socket to `HNode` to worry about.

TLDR; No support yet. Experienced developer can do it themselves suprisingly easily.
